Userland/Utilities/acpi.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022-2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
+#include <LibCore/System.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// https://uefi.org/specs/ACPI/6.4/05_ACPI_Software_Programming_Model/ACPI_Software_Programming_Model.html#system-description-table-header
+struct [[gnu::packed]] SDTHeader {
+    char sig[4];
+    u32 length;
+    u8 revision;
+    u8 checksum;
+    char oem_id[6];
+    char oem_table_id[8];
+    u32 oem_revision;
+    u32 creator_id;
+    u32 creator_revision;
+};
+
+static ErrorOr<NonnullOwnPtr<Core::Stream::BufferedFile>> open_file_or_stdin(String const& filename)
+{
+    OwnPtr<Core::Stream::File> file;
+    if (filename == "-") {
+        file = TRY(Core::Stream::File::adopt_fd(STDIN_FILENO, Core::Stream::OpenMode::Read));
+    } else {
+        file = TRY(Core::Stream::File::open(filename, Core::Stream::OpenMode::Read));
+    }
+    return TRY(Core::Stream::BufferedFile::create(file.release_nonnull()));
+}
+
+static void pkg_length(Vector<u8> const& data, u32& next_in_block, u32& next_block)
```

Comment:
You should probably have a look at the [CodingStyle.md](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md) as it turns out I can't remember simultaneously more than one rule

Something more expressive and without abbreviation would be a better fit, maybe `print_package`?

Reply:
I prefere to not change the name since it is from the ACPI specification.
The output is just debug messages, and will be removed in a later commit

Reply:
Wasn't aware of this. Probably fine, then